src/koop/tokenMarkLogicAuthentication.js

```diff
@@ -17,85 +17,65 @@ function tokenMarkLogicAuthentication(options) {
             if (err) {
-                log.debug(err);
-                err.code = 403;
-                reject(err);
+              log.debug(`Token validation failed: ${JSON.stringify(err)}`);
+              err.code = 401;
+              reject(err);
             }
-            let username = decoded.sub;
+            const username = decoded.sub;
             req.markLogicClientCacheKey = username;
 
-            //now we need to make sure the client didn't get booted out of cache
+            // Verify that the client in the client cache still exists. If it expired, then force the user to
+            // re-authenticate.
             if (dbClientManager.getCachedMarkLogicClient(username)) {
-                log.debug("authorize successful: dbClient previusly connected");
-                // Resolve the decoded token (an object)
-                resolve(decoded);
-            }
-            else {
-                //if it did get booted out, we can't talk to ML until the
-                //client re-authenticates; reject with a 401
-                reject({message:"Please re-authenticate", code:401});
+              resolve(decoded);
+            } else {
+              reject({code: 401});
             }
-        });
+          });
+        }
     });
 }
 
 function validateCredentials(req, username, password, resolve, reject) {
-    dbClientManager.connectAndCacheClient(username, password).then(response => {
-    log.debug("response from connectClient: ");
-    log.debug(response);
+  dbClientManager.connectAndCacheClient(username, password).then(response => {
     if (response.authenticated) {
-        log.debug("successfully authenticated " + username);
-        let expiration = _tokenExpirationMinutes;
-        // Create access token and wrap in response object
-        if (req && req.query) {
-            expiration = req.query.expiration || _tokenExpirationMinutes;
-            if (expiration > _tokenExpirationMinutes)
-                expiration = _tokenExpirationMinutes;
+      let expiration = _tokenExpirationMinutes;
+      // Create access token and wrap in response object
+      if (req && req.query) {
+        expiration = req.query.expiration || _tokenExpirationMinutes;
+        if (expiration > _tokenExpirationMinutes) {
+          expiration = _tokenExpirationMinutes;
         }
-        log.debug("about to make jwt");
-        let expires = Date.now() + (expiration * 60 * 1000);
-        let json = {
-            token: jwt.sign({exp: Math.floor(expires / 1000), sub: username}, _secret),
-            expires
-        };
-        req.markLogicClientCacheKey = username;
-        resolve(json);
-    } else if (response.authenticated == false) {
-        let err = new Error('Invalid credentials.');
-        err.code = 401;
-        err.httpStatusMessage = response.httpStatusMessage;
-        err.httpStatusCode = response.httpStatusCode;
-        reject(err);
-    }
-    else {
-        let err = new Error("MarkLogic error");
```

Comment:
This code was never reached - the connectAndCacheClient method either returns a "response" object via `resolve` or it tosses an error via `reject`, which is handled by the `catch` method.